gpt_index/callbacks/schema.py

```diff
@@ -0,0 +1,47 @@
+"""Base schema for callback managers."""
+import uuid
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, Optional
+
+TIMESTAMP_FORMAT = "%m/%d/%Y, %H:%M:%S"
+
+
+class CBEventType(str, Enum):
+    """Callback manager event types."""
+
+    CHUNKING = "chunking"
+    EMBEDDING = "embedding"
+    LLM = "llm"
+    # TODO: Can we use these anywhere?
+    # QUERY = "query"
+    # RETRIEVE = "retrieve"
+    # SYNTHESIZE = "synthesize"
+    TREE = "tree"

```

Comment:
curious: why does tree need its own event type?

Reply:
Tree's are a little more unique in what gets logged/tracked (i.e. summaries of different levels), so I thought it made sense to give it it's own event type.

Are you thinking it just be part of the LLM events?

Reply:
I see, that makes sense.  I was imagining logging would be the more lower level primitives, but this is fine as well.

Reply:
I think we are going to deprecate tree index soon, so doesn't matter too much.